Detour/Source/DetourNavMesh.cpp

```diff
@@ -639,41 +675,26 @@ void dtNavMesh::closestPointOnPoly(dtPolyRef ref, const float* pos, float* close
 	const unsigned int ip = (unsigned int)(poly - tile->polys);
 	const dtPolyDetail* pd = &tile->detailMeshes[ip];
 	
-	// Clamp point to be inside the polygon.
 	float verts[DT_VERTS_PER_POLYGON*3];	
-	float edged[DT_VERTS_PER_POLYGON];
-	float edget[DT_VERTS_PER_POLYGON];
 	const int nv = poly->vertCount;
 	for (int i = 0; i < nv; ++i)
 		dtVcopy(&verts[i*3], &tile->verts[poly->verts[i]*3]);
 	
-	dtVcopy(closest, pos);
-	if (!dtDistancePtPolyEdgesSqr(pos, verts, nv, edged, edget))
+	if (!dtPointInPolygon(pos, verts, nv))
 	{
-		// Point is outside the polygon, dtClamp to nearest edge.
-		float dmin = edged[0];
-		int imin = 0;
-		for (int i = 1; i < nv; ++i)
-		{
-			if (edged[i] < dmin)
-			{
-				dmin = edged[i];
-				imin = i;
-			}
-		}
-		const float* va = &verts[imin*3];
-		const float* vb = &verts[((imin+1)%nv)*3];
-		dtVlerp(closest, va, vb, edget[imin]);
+		// Clamp to nearest detail edge.
+		closestPointOnDetailEdge(tile, poly, pos, closest);
```

Comment:
This should work, but I'd test that it returns same-ish point as previous code.

Reply:
I wonder if this is even better since we are now unconditionally iterating all detail tri edges, while before we had an early out (in code below) if we found the containing triangle. I wonder which is better (with no epsilon it is probably likely that all the triangle checks below fail, however).

Reply:
I like this behavior better, since this is common case where the raycasts fail. My point was that just make sure the `closestPointOnDetailEdge` returns similar results as previous code since it is using different data. I believe it should work, but just double check :)